crates/ruff_cli/src/commands/format.rs

```diff
@@ -567,3 +572,69 @@ impl Display for FormatCommandError {
         }
     }
 }
+
+pub(super) fn warn_incompatible_formatter_settings(
+    pyproject_config: &PyprojectConfig,
+    resolver: Option<&Resolver>,
+) {
+    for setting in std::iter::once(&pyproject_config.settings)
+        .chain(resolver.iter().flat_map(|resolver| resolver.settings()))
+    {
+        let mut incompatible_rules = Vec::new();
+
+        for incompatible_rule in RuleTable::from_iter([
+            Rule::LineTooLong,
+            Rule::TabIndentation,
+            Rule::IndentationWithInvalidMultiple,
+            Rule::IndentationWithInvalidMultipleComment,
+            Rule::OverIndented,
+            Rule::IndentWithSpaces,
+            Rule::SingleLineImplicitStringConcatenation,
+            Rule::MissingTrailingComma,
+            Rule::ProhibitedTrailingComma,
+            Rule::BadQuotesInlineString,
+            Rule::BadQuotesMultilineString,
+            Rule::BadQuotesDocstring,
+            Rule::AvoidableEscapedQuote,
+        ])
+        .iter_enabled()
+        {
+            if setting.linter.rules.enabled(incompatible_rule) {
+                incompatible_rules.push(format!("'{}'", incompatible_rule.noqa_code()));
+            }
+        }
+
+        if !incompatible_rules.is_empty() {
+            incompatible_rules.sort();
+            warn!("The following rules can cause conflicts when used with the formatter: {}. We recommend disabling these rules when using Ruff's formatter to avoid unexpected behavior. To disable the rules, change your configuration and either remove the rules from the `select` or `extend-select` options or add the rules to the `ignore` option.", incompatible_rules.join(", "))
+        }
+
+        let mut incompatible_options = Vec::new();
+
+        let isort_defaults = isort::settings::Settings::default();
+
+        if setting.linter.isort.force_single_line != isort_defaults.force_single_line {
+            incompatible_options.push("'isort.force-single-line'");
+        }
+
+        if setting.linter.isort.force_wrap_aliases != isort_defaults.force_wrap_aliases {
+            incompatible_options.push("'isort.force-wrap-aliases'");
+        }
+
+        if setting.linter.isort.lines_after_imports != isort_defaults.lines_after_imports {
+            incompatible_options.push("'isort.lines-after-imports'");
+        }
+
+        if setting.linter.isort.lines_between_types != isort_defaults.lines_between_types {
+            incompatible_options.push("'isort.lines_between_types'");
+        }
+
+        if setting.linter.isort.split_on_trailing_comma != isort_defaults.split_on_trailing_comma {
+            incompatible_options.push("'isort.split_on_trailing_comma'");
+        }
+
+        if !incompatible_options.is_empty() {
+            warn!("The following isort options are incompatible with the formatter: {}. We recommend disabling these isort options when using Ruff's formatter by removing them from the configuration.", incompatible_options.join(", "))
```

Comment:
Small nudges on the comment:

```
The following isort options may cause conflicts when used with the formatter: {}.
To avoid unexpected behavior, we recommend disabling these options by removing
them from your configuration.
```

Reply:
Nice